addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,115 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import re
+import requests
+from requests.exceptions import HTTPError, Timeout
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductGetPicUsingBarcode(models.TransientModel):
+    """ Get product pictures using barcode on Google Image
+
+    :raises UserError: If the user didn't enter API Key or Search Engine ID in the settings
+    :raises UserError: If there isn't any product without a picture and with a barcode selected.
+    """
+    _name = 'product.images.get.pic.barcode'
+    _description = 'Get Picture with Barcode Number'
+
+    @api.model
+    def default_get(self, fields):
+        default = super().default_get(fields)
+
+        google_custom_search_cx = bool(self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.cx'))
+        google_custom_search_key = bool(self.env['ir.config_parameter'] \
+            .get_param('google.custom_search.key'))
+
+        if not google_custom_search_cx or not google_custom_search_key:
+            raise UserError(_("You didn't configure properly your API Key or Search Engine ID."))
+
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.template'].browse(self._context.get('active_ids')) \
+                .product_variant_ids.filtered(lambda p: not p.image_1920 and p.barcode)
+        else:
+            product_ids = self.env['product.product'].browse(self._context.get('active_ids')) \
+                .filtered(lambda p: not p.image_1920 and p.barcode).ids
+
+        if len(product_ids) == 0:
+            raise UserError(_("There isn't any product without a picture and with a barcode."))
+        # TODO: Define a maximum number of product to process in the same batch
+        if len(product_ids) > 100:
+            raise UserError(_("You selected too many product variants at the same time. \
+Please select a maximum of 100 product variants in the same batch."))
+
+        default.update({
+            'product_ids': product_ids,
+            'nb_product': len(product_ids),
+        })
```

Comment:
```suggestion
        default['product_ids'] = product_ids
        default['nb_product'] = len(product_ids)
```
even if you absolutely want to use `dict.update`, there's no cause to call it with a dict literal, just use kwargs.